src/error.rs

```diff
@@ -0,0 +1,154 @@
+use crate::qdrant::value::Kind::*;
+use crate::qdrant::{ListValue, Struct, Value};
+use std::error::Error;
+use std::fmt::{Debug, Display, Formatter};
+use std::marker::PhantomData;
+
+/// An error for failed conversions (e.g. calling `String::try_from(v)`
+/// on an integer [`Value`](Value))
+pub struct NotA<T> {
```

Comment:
In the end we probably want to remove this, as I don't see a clear reason for having it.

But that would of course be at  the end of the whole rewrite.